### PR TITLE
chore: Improve the http body check in e2e test case

### LINKED ISSUE #1246: chore: change the type of `ExpectBody`  to []string in backend e2e test
# Feature request

## Please describe your feature

Hi, now the type of `ExpectBody` is `string` in the backend e2e test. 
See: https://github.com/apache/apisix-dashboard/blob/40d8f97de8770187599ff5963617049603020e92/api/test/e2e/base.go#L177

Sometimes we need to check multiple strings (such as the list function) in the response body. So I recommend changing the type of `ExpectBody` to `[]string`.

## Describe the solution you'd like

Change the type of `ExpectBody` to `[]string`.

### PR DESCRIPTION
**Pull request by imjoey**
Signed-off-by: imjoey <majunjiev@gmail.com>

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

Fixes #1246 .

___

Changes proposed in this PR:

1. Change the name of `UnexpectedBody` property to `UnexpectBody` to get along with `ExpectBody` field;
2. Set the types of `ExpectBody` and `UnexpectBody ` to `interface{}`, to add support for checking on multiple strings. See #1246 for details. Also use this new feature to refactor the test case of `server_info_test.go` to reduce time cost;
3. Fix a logical bug. Previously, if the `ExpectBody` is set as `""`, the check will be certainly ignored. While actually I believe this purpose is to expect the body be empty. Please see https://github.com/apache/apisix-dashboard/blob/93b9765e3e77c6cefe0487e1657886d5f05d4146/api/test/e2e/route_with_plugin_cors_test.go#L138-L153 as an example;
4. Other minor style fixes;

**Note:**
Since the changes above are all minor, and related to http body check in e2e testing, I put them in this single PR. Please let me know if they should be put into different PRs, respectively. Thanks.

### CODE REVIEW

**Comment by imjoey** [comment]
ping @starsz @juzhiyuan for review. Any feedback is welcome. Much appreciated.

File: api/test/e2e/server_info_test.go [hunk 1]
```diff
@@ -121,25 +112,14 @@ func TestServerInfo_Get_OmitEmptyValue(t *testing.T) {
...
-		{
-			Desc:           "list server info with hostname",
-			Object:         ManagerApiExpect(t),
-			Path:           "/apisix/admin/server_info",
-			Query:          "hostname=apisix_",
-			Method:         http.MethodGet,
-			Headers:        map[string]string{"Authorization": token},
-			ExpectStatus:   http.StatusOK,
-			ExpectBody:     "\"total_size\":2",
-			UnexpectedBody: "\"update_time\":",
```
Comment (starsz) [inline api/test/e2e/server_info_test.go:142]: Hi, why delete this test?
This test is for `list server info with hostname`.

Comment (imjoey) [reply api/test/e2e/server_info_test.go:142]: @starsz oh, sorry, my bad. Will restore it soon.

Comment (imjoey) [reply api/test/e2e/server_info_test.go:142]: @starsz PR renewed with restoring the mistakenly removed code, much appreciated for your reviewing.

**Review by starsz — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Comment by juzhiyuan** [comment]
This PR will get merged after https://github.com/apache/apisix-dashboard/pull/1272

**Comment by juzhiyuan** [comment]
No need to retrigger Update branch :)

**Comment by imjoey** [comment]
> No need to retrigger Update branch :)

@juzhiyuan OK. Thanks.

**Pull request merged**